### ISSUE TITLE
GPU support for evaluation

### ISSUE THREAD

**Issue by alainjungo** [rundherum/pymia#33]
For future releases, GPU support for evaluation might be worth considering to speed up validaion during training.

**Comment by neuronflow** [comment]
I second this, wanted to request the same 👍

**Comment by fabianbalsiger** [comment]
@neuronflow contributions are always welcome. To add a bit more information:

Currently, the evaluation is implemented in NumPy and, therefore, not compatible with GPU use. Implementing the evaluation in TensorFlow and PyTorch would be possible. However, this breaks our paradigm of being framework-independent and further adds overhead for unit testing (which should be the main focus for now regarding the metrics). Possibly, Python packages as CuPy and JAX could be used to compile the NumPy code for GPU use. Further, the loop in the evaluation of the subjects would need to be parallelized to gain a significant speed increase over non-GPU execution. It is, therefore, questionable if a speed gain is even possible (considering also the copy of the data to the GPU and back).

Implementing CuPy JAX compilation for one metric and benchmarking this to plain NumPy would be a good start to get a glimpse of what would be possible.
